client/blocks/plan-storage/index.jsx

```diff
@@ -55,18 +55,30 @@ export class PlanStorage extends Component {
 		const planHasTopStorageSpace =
 			isBusinessPlan( sitePlanSlug ) || isEcommercePlan( sitePlanSlug );
 
-		return (
-			<div className={ classNames( className, 'plan-storage' ) }>
+		const displayUpgradeLink = canUserUpgrade && ! planHasTopStorageSpace;
+
+		const planStorageComponents = (
+			<>
 				<QueryMediaStorage siteId={ siteId } />
 				<PlanStorageBar
-					siteSlug={ siteSlug }
 					sitePlanSlug={ sitePlanSlug }
 					mediaStorage={ this.props.mediaStorage }
-					displayUpgradeLink={ canUserUpgrade && ! planHasTopStorageSpace }
+					displayUpgradeLink={ displayUpgradeLink }
 				>
 					{ this.props.children }
 				</PlanStorageBar>
-			</div>
+			</>
+		);
+
+		if ( displayUpgradeLink ) {
+			return (
+				<a className={ classNames( className, 'plan-storage' ) } href={ `/plans/${ siteSlug }` }>
+					{ planStorageComponents }
+				</a>
+			);
+		}
+		return (
+			<div className={ classNames( className, 'plan-storage' ) }>{ planStorageComponents }</div>
```

Comment:
I'll leave this up to you, but we can decompose this further by adding a wrapper component. (There's also more than one way to do this, so feel free to experiment. This is non blocking, but I find it's a bit easier to read.

```jsx
//psuedocode
const planStorageContainer = ( { children, displayUpgradeLink, planHref } ) => {
    const tagName = displayUpgradeLink ? 'a' : 'div' 
    //an if <a /> else <div/> also works
    return React.createElement( tagName, 
         { className: classNames( className, 'plan-storage' ),  ...( displayUpgradeLink && { href: planHref } ) }, 
         children
     );
}

//...planStorage
return (
    <PlanStorageContainer>
        <QueryMediaStorage siteId={ siteId } />
				<PlanStorageBar
					siteSlug={ siteSlug }
					sitePlanSlug={ sitePlanSlug }
					mediaStorage={ this.props.mediaStorage }
					displayUpgradeLink={ canUserUpgrade && ! planHasTopStorageSpace }
				>
					{ this.props.children }
				</PlanStorageBar>
    </PlanStorageContainer>
```

Reply:
It's funny, I tried the wrapper component approach first, but I must have done something wrong because it kept re-rendering -- the wrapper div kept flashing in the inspector as though it had changed, and I couldn't target it. :D I wasn't using createElement, though, so maybe that has something to do with it!

Reply:
> the wrapper div kept flashing in the inspector as though it had changed, and I couldn't target it.

Sounds like something was triggering a re-render, for example a new object reference passed through to props (even if content of that object didn't change).

Reply:
My wrapper note is totally optional btw 👍